test/integration/services/dealer/hedge-price.spec.ts

```diff
@@ -0,0 +1,1892 @@
+  const payResult = await Payments.payInvoiceByWalletId({
+    uncheckedPaymentRequest: lnInvoice.paymentRequest,
+    memo: null,
+    senderWalletId: newUsdWallet.id,
+    senderAccount: newAccount,
+  })
+  if (payResult instanceof Error) throw payResult
+  const afterUsd = await getBalanceHelper(newUsdWallet.id)
+  const diff = (beforeUsd - afterUsd) as CurrencyBaseAmount
+  return diff
+}
+
+const getBtcForUsdEquivalentNoAmountInvoice = async ({
+  btcPaymentAmount,
+  accountAndWallets,
+}: {
+  btcPaymentAmount: BtcPaymentAmount
+  accountAndWallets: AccountAndWallets
+}): Promise<CurrencyBaseAmount | ZeroAmountForUsdRecipientError> => {
+  const { newBtcWallet, newUsdWallet, newAccount } = accountAndWallets
+
+  const lnInvoice = await Wallets.addInvoiceNoAmountForSelf({
+    walletId: newUsdWallet.id,
+  })
+  if (lnInvoice instanceof Error) throw lnInvoice
+
+  const beforeBtc = await getBalanceHelper(newBtcWallet.id)
+  const result = await Payments.payNoAmountInvoiceByWalletId({
+    amount: Number(btcPaymentAmount.amount),
+    uncheckedPaymentRequest: lnInvoice.paymentRequest,
+    memo: null,
+    senderWalletId: newBtcWallet.id,
+    senderAccount: newAccount,
+  })
+  if (result instanceof Error) {
+    if (!(result instanceof ZeroAmountForUsdRecipientError)) throw result
+    return result
+  }
+  const afterBtc = await getBalanceHelper(newBtcWallet.id)
+  const diff = (beforeBtc - afterBtc) as CurrencyBaseAmount
+  return diff
+}
+
+const getBtcForUsdEquivalentNoAmountInvoiceAndProbe = async ({
+  btcPaymentAmount,
+  accountAndWallets,
+}: {
+  btcPaymentAmount: BtcPaymentAmount
+  accountAndWallets: AccountAndWallets
+}): Promise<CurrencyBaseAmount | ZeroAmountForUsdRecipientError> => {
+  const { newBtcWallet, newUsdWallet, newAccount } = accountAndWallets
+
+  const lnInvoice = await Wallets.addInvoiceNoAmountForSelf({
+    walletId: newUsdWallet.id,
+  })
+  if (lnInvoice instanceof Error) throw lnInvoice
+
+  const beforeBtc = await getBalanceHelper(newBtcWallet.id)
+
+  const probe = await Payments.getNoAmountLightningFeeEstimation({
+    amount: Number(btcPaymentAmount.amount),
+    uncheckedPaymentRequest: lnInvoice.paymentRequest,
+    walletId: newBtcWallet.id,
+  })
+  if (probe instanceof Error) {
+    if (!(probe instanceof ZeroAmountForUsdRecipientError)) throw probe
+    return probe
+  }
+
+  const result = await Payments.payNoAmountInvoiceByWalletId({
+    amount: Number(btcPaymentAmount.amount),
+    uncheckedPaymentRequest: lnInvoice.paymentRequest,
+    memo: null,
+    senderWalletId: newBtcWallet.id,
+    senderAccount: newAccount,
+  })
+  if (result instanceof Error) {
+    if (!(result instanceof ZeroAmountForUsdRecipientError)) throw result
+    return result
+  }
+  const afterBtc = await getBalanceHelper(newBtcWallet.id)
+  const diff = (beforeBtc - afterBtc) as CurrencyBaseAmount
+  return diff
+}
+
+const getMaxBtcAmountToEarn = async ({
+  startingBtcAmount,
+  accountAndWallets,
+  diffFn,
```

Comment:
I guess this is not just diff, it's a no change comment and I know we are using this in src but this introduces an overhead in logic and is really hard to track, especially if the fn name doesn't include what the function actually does

Reply:
With the new names does this function make more sense now? I tried to explain what the function does with comments throughout the function as well. It executes a 3 step strategy to find the max BTC amount one can use to pay an invoice or recipient wallet.